src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkTask.java

```diff
@@ -125,6 +132,39 @@ private void checkMapping(SinkRecord record) {
     }
   }
 
+  private String getServerVersion() {
+    ConfigCallbackHandler configCallbackHandler = new ConfigCallbackHandler(config);
+    RestHighLevelClient highLevelClient = new RestHighLevelClient(
+        RestClient
+            .builder(
+                config.connectionUrls()
+                    .stream()
+                    .map(HttpHost::create)
+                    .collect(Collectors.toList())
+                    .toArray(new HttpHost[config.connectionUrls().size()])
+            )
+            .setHttpClientConfigCallback(configCallbackHandler)
+            .setRequestConfigCallback(configCallbackHandler)
+    );
+    MainResponse response;
+    String esVersionNumber = "Unknown";
+    try {
+      response = highLevelClient.info(RequestOptions.DEFAULT);
+      esVersionNumber = response.getVersion().toString();
+    } catch (Exception e) {
+      // Same error messages as from validating the connection for IOException.
+      // Insufficient privileges to validate the version number if caught
+      // ElasticsearchStatusException.
+      log.warn("Failed to get ES server version", e);
+    }
+    try {
+      highLevelClient.close();
+    } catch (Exception e) {
+      log.error("Failed to close high level client");
```

Comment:
warn

Reply:
also put this in finally of the above try?

Reply:
Shouldn't it be an error? Since leaving a client open could cause issues if it happens many times over the life of a cluster.

Reply:
I was thinking that leaving the client open could be potentially problematic over the long life span of a cluster